src/main/java/com/vi/migrationtool/mongodb/DropLockedAgenciesAndSaveDataAtrribute.java

```diff
@@ -0,0 +1,45 @@
+package com.vi.migrationtool.mongodb;
+
+import com.vi.migrationtool.common.MigrationTasks;
+import com.vi.migrationtool.config.BeanAwareSpringLiquibase;
+import com.vi.migrationtool.mongodb.model.ConsultingTypeEntity;
+import com.vi.migrationtool.mongodb.service.ConsultingTypeService;
+import java.util.List;
+import liquibase.database.Database;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class DropLockedAgenciesAndSaveDataAtrribute extends MigrationTasks {
+
+  @Override
+  public void execute(Database database) {
+    ConsultingTypeService consultingTypeService =
+        BeanAwareSpringLiquibase.getBean(ConsultingTypeService.class);
+    migrate(consultingTypeService);
+  }
+
+  public void migrate(ConsultingTypeService consultingTypeService) {
+    List<ConsultingTypeEntity> consultingTypeEntities = consultingTypeService.getConsultingTypes();
+    applyMigration(consultingTypeService, consultingTypeEntities);
+  }
+
+  private void applyMigration(
+      ConsultingTypeService consultingTypeService,
+      List<ConsultingTypeEntity> consultingTypeEntities) {
+    if (consultingTypeEntities != null) {
+      consultingTypeEntities.stream()
+          .forEach(
+              (consultingTypeEntity) ->
+                  applyMigration(consultingTypeEntity, consultingTypeService));
+    } else {
+      log.info("Skipping migration as consulting type entities are null");
+    }
+  }
+
+  private void applyMigration(
+      ConsultingTypeEntity consultingTypeEntity, ConsultingTypeService consultingTypeService) {
+    consultingTypeEntity.setLockedAgencies(null);
```

Comment:
what is the result of this migration?
In mongo DB there should be entries without "lockedAgencies" attribute, not with lockedAgencies set to null
I think in order to remove it like this, you'd need to remove this attribute lockedAgencies from ConsultingTypeEntity, load all entities and then persist them again.

Reply:
The question is providing the latest schema alone should take care the objects that have the locked agencies and savedatamessage attribute or we should nulliffy them and then fix the schema?